pkg/driver/node.go

```diff
@@ -791,8 +791,12 @@ func (d *NodeService) getVolumesLimit() int64 {
 		availableAttachments = dedicatedLimit
 	} else if isNitro {
 		enis := d.metadata.GetNumAttachedENIs()
-		nvmeInstanceStoreVolumes := cloud.GetReservedSlotsForInstanceType(instanceType)
-		availableAttachments = availableAttachments - enis - nvmeInstanceStoreVolumes
+		reservedSlots := cloud.GetReservedSlotsForInstanceType(instanceType)
+		if hasMaxvolumeLimit {
+			availableAttachments = availableAttachments - reservedSlots
```

Comment:
```suggestion
			availableAttachments = availableAttachments - (enis - 1) - reservedSlots
```

I'm pretty sure this is what we actually should do here, as despite being special cases these devices still are a shared limit.

Reply:
Ah yes this is better as if an instance has more than one eni it does count towards the shared limit will fix on next rev thank you !